presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/readers/SliceColumnReader.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.readers;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.VariableWidthBlock;
+import com.facebook.presto.thrift.interfaces.client.ThriftColumnData;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+final class SliceColumnReader
+{
+    private SliceColumnReader()
+    {
+    }
+
+    private static void checkConsistency(boolean[] nulls, byte[] bytes, int[] sizes, int totalRecords)
```

Comment:
Unfortunately, I didn't find a good way to do it.
First, the check is specific to each reader. Second, you cannot use generics with primitive arrays in this case, because primitive array cannot be passed as parameters like `T[]`.
I can de-duplicate nulls checks (first 2 conditions in most readers), but it doesn't seem to have enough value.